nexus/db-queries/src/db/datastore/region_snapshot_replacement.rs

```diff
@@ -1315,10 +1370,18 @@ mod test {
             .await
             .unwrap();
 
-        datastore
+        let result = datastore
             .insert_region_snapshot_replacement_step(&opctx, step.clone())
             .await
-            .unwrap_err();
+            .unwrap();
+
+        let InsertRegionSnapshotReplacementStepResult::AlreadyHandled {
+            existing_step_id,
+        } = result
+        else {
+            panic!("wrong return type");
```

Comment:
similarly, might be good for this panic to show what the wrong thing was?

Reply:
yep: d2135f21b